Multiple omero tables

will-moore opened the pull request:
Fixes #351

This adds support for multiple OMERO.tables on the containers of an Image or Well. See issue above.

To test:

 - For the cases where we only have a single table on a image container (Screen, Plate, Project or Dataset) this table will be loaded immediately: e.g. https://merge-ci.openmicroscopy.org/web/webclient/?show=well-10584 (user-3)

![Screenshot 2022-02-07 at 17 14 17](https://user-images.githubusercontent.com/900055/152838554-7cc308f7-c412-4668-9b58-f7c8d1b85011.png)

 - When we have multiple tables, they are listed and can each be loaded by clicking on them. e.g. https://merge-ci.openmicroscopy.org/web/webclient/?show=image-18221 (user-3):

![Screenshot 2022-02-07 at 17 18 23](https://user-images.githubusercontent.com/900055/152838657-4d3fc49d-ebc3-49bc-b04c-1927474a3837.png).

In both cases, the tooltip on the table gives extra info as before.

cc @sbesson @francesw

sbesson reviewed (commented):
The IDR use case behind this PR certainly points the current UI behavior as not fit for purpose for dealing with multiple tabular attachments.

Overall I find the new UI element (replacing the `Info` element) to be a natural and straighforward way to expose multiple tables. It is consistent with the UI used for `Attachments`. The auto-expansion feature in the case of single tables does not break the user workflow and for multi-tables, it allows to expand/shrink/navigate between the available tables.

One inline question about the tooltip. Otherwise, two general comments:

- the `(linked to: )` text is interesting as it provides some provenance metadata. I wonder whether this information should necessarily be visible immediately to the user. Naively, I was almost expecting clicking on `Project: xxx` to bring me to this project. An alternative would be to move it under the tooltip 
- with more use cases, I can foresee some RFEs associated with grouping these annotations by namespaces

Including also the Glencoe team to make sure this UI addition would not break existing workflow @knabar @erindiel (anyone else? 😄 )

will-moore commented:
I fixed the tooltip: "Linked by".
I also updated the `(Linked to Project-1)` to actually be a link.
But I think that it is useful to have that at the top-level on the object itself instead of in the tooltip as it helps give a better understanding of where these are coming from: to understand that these are not linked to the Image itself. E.g. see confusion at https://forum.image.sc/t/omero-saving-as-omero-table/54360/3

I also added sorting by name then date, so if you have a bunch of tables, they are sorted consistently:
<img width="362" alt="Screenshot 2022-02-08 at 23 02 46" src="https://user-images.githubusercontent.com/900055/153162904-382ec2e4-ea09-42b9-8096-ad41ed417c72.png">

sbesson commented:
Thanks, the tooltip looks good now. The only downside of the latest version is that clicking on the triangle icon  does not expand/shrink the table. Instead you need to click on the table name. Is this expected?

![Screenshot 2022-02-09 at 09 16 52](https://user-images.githubusercontent.com/1355463/153163900-6b982ea3-9448-4918-9e17-3e95e87d8e53.png)

will-moore commented:
Hmmm, In order to make the link clickable as a link, I had to remove the `[click the whole row to expand]` behaviour.
But I'll try and fix it so that clicking on the name OR the `>` does the expanding

will-moore commented:
Now showing Tables counts for Wells and Images:

![Screenshot 2022-02-09 at 16 09 33](https://user-images.githubusercontent.com/900055/153241931-337ebbd2-f6cd-44e5-afdc-a2da8c494df4.png)

sbesson reviewed (approved):
With the last commit, the clickable issue raised above is now fixed. The addition of `Tables` count is also a great asset for end users to be able to know that one or several tables are associated to a particular image and facilitate the discovery of this metadata

I assume the extra queries performed to compute the tables count will have a minimal performance impact but this is something we might need to check on an IDR scale deployment. 

Otherwise, I find the proposed changes brings the handling of tabular metadata in the right-hand panel to parity with image annotations and goes in the sense of increased usability. Happy to see this included in the next minor release of OMERO.web.
